### PR TITLE
Backend workflow

### PR DESCRIPTION
**Pull request by shabbyblue16**
## Description
Attempted to organize the backend in a way that hopefully simplifies and reduces friction. Everything's got a little home in the country of our code. Open to any critique and happy to modify accordingly. There's a bunch of dummy code, empty folders don't push to github. Feel free to delete.

### Backend Workflow
1. Database schema lives in `server/models/`. Each component has its own file but they may not all be necessary (e.g. not sure the navBar will need its own data models). Create any Schema and corresponding Model and add the Model to the `module.exports` object.
2. Route handlers exist in `server/controllers/`. Import any needed Models. Define functions with any db queries and add to the `module.exports` object.
3. Routes are in `routes/routes.js`. Add routes with corresponding controller function. Example: `router.get('/friends/search', friendsList.search)`.
4. Any middleware we might need will be added to `server/app.js`

## How to test
`npm start` and confirm connection to database

## Notes
- I've added dependencies in the client folder which is probably better practice. Louisa made a good point about messing with node_modules being a little hacky (which I think is fair).
- `npm install` from the root directory and also from the client directory.
- `npm start` from the root directory, and `expo start` from the client
-  Ensure mongo database is running (if the server doesn't run this may be why).
- Create `DB_URI` environmental variable with the mongodb uri (with localhost for now, ec2 ip after deployment) in a .env file.
- Database name is 'quick-bagel'
- Double check that client dependency versions line up with these:
```
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "react-native": "0.64.3",
    "react-native-gesture-handler": "~1.10.2",
    "react-native-reanimated": "~2.2.0",
    "react-native-safe-area-context": "^3.3.2",
    "react-native-screens": "~3.8.0",
    "react-native-web": "0.17.1"
```
I ran into errors and had to `expo install` to get the correct versions to make expo happy

## Issue tracking
https://trello.com/c/Gd6h9PNd/8-skeleton-code-setup

### CODE REVIEW

File: server/models/dashboard.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
```
Comment (Laweeza) [inline server/models/dashboard.js:4]: I like how you created separate model files to separate the schemas, more appropriate for the anticipated app size.

File: server/routes/routes.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+
+const {
```
Comment (Laweeza) [inline server/routes/routes.js:3]: I do like this better vs exporting all of the controllers into an index.js file.

**Comment by Laweeza** [comment]
[Skeleton code setup](https://trello.com/c/Gd6h9PNd/8-skeleton-code-setup)

**Review by ryhorowitz — APPROVED** [review]
Justin,
Looks good to me. Thank you for the well written instructions!

**Review by Laweeza — APPROVED** [review]

**Review by aaronfife — APPROVED** [review]
you rock Justin!

**Pull request merged**